### PR TITLE
Add learner-profile-dashboard role.

### PR DESCRIPTION
**Pull request by tomaszgy**
This PR adds learner-profile-dashboard for provisioning a stage server of the Learner Profile Dashboard.
cf [OC-4364](https://tasks.opencraft.com/browse/OC-4364)

The role was used to deploy https://lpd-stage.opencraft.hosting

Squashed from ddd507033c04bb42036dca853a1a66e303c29611.

Signed-off-by: Tomasz Gargas <tomasz@opencraft.com>

### CODE REVIEW

File: playbooks/roles/learner-profile-dashboard/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+LPD_LOG_DOWNLOAD_AUTHORIZED_KEYS: []
+LPD_LOG_DOWNLOAD_USER_HOME: "/home/{{ LPD_LOG_DOWNLOAD_USER }}"
+LPD_LOG_DOWNLOAD_LOG_DIR: "{{ LPD_LOG_DOWNLOAD_USER_HOME }}/logs"
+LPD_LOG_DOWNLOAD_DB_DIR: "{{ LPD_LOG_DOWNLOAD_USER_HOME }}/db"
+LPD_LOG_TARSNAP_ARCHIVE: lpd-logs
+
+LPD_LOG_DOWNLOAD_VOLUME_DEVICE_ID: /dev/vdc
+LPD_LOG_DOWNLOAD_VOLUME_FILESYSTEM: ext4
+
+# Note: when overriding these leave DJANGO_SETTINGS_MODULE inside
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/defaults/main.yml]: @tomaszgy You might have copied this from somewhere else (?), but I'm not sure what `leave DJANGO_SETTINGS_MODULE inside` means?

File: playbooks/roles/learner-profile-dashboard/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  DJANGO_SETTINGS_MODULE: 'app.settings'
+
+LPD_LOG_DIR: "/var/log/lpd"
+
+LPD_STATICFILES_ROOT: "{{ LPD_USER_HOME }}/static"
+LPD_MEDIA_ROOT: "{{ LPD_USER_HOME }}/media"
+
+LPD_REPOSITORY: https://github.com/open-craft/learner-profile-dashboard.git
+
+# Temporary commit containing LPD changes
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/defaults/main.yml]: @tomaszgy If `LPD_REPOSITORY_VERSION` points to `master`, do we still need this comment?

File: playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+- name: install dependencies
+  pip:
+    virtualenv: "{{ LPD_VENV }}"
+    state: present
+    requirements: "{{ LPD_APP_LOCATION }}/requirements/base.txt"
+    virtualenv_python: "/usr/bin/python2.7"
+
+# This is a hack Gunicorn==17.5 is installed system wide, so virtualenv and system version match.
+# possibly this should be sloved in a cleaner way
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml]: @tomaszgy This comment should mention why we need a different version of `Gunicorn`.

Also, a couple typos to fix are:

* Missing `:` after `This is a hack`
* Capitalize `possibly`
* `sloved` should be `solved` 😄

File: playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- name: install dependencies
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml]: @tomaszgy Change to `install gunicorn` to be more specific?

File: playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  pip:
+    virtualenv: "{{ LPD_VENV }}"
+    state: present
+    name: "Gunicorn==19.4.5"
+    virtualenv_python: "/usr/bin/python2.7"
+
+- name: install dependencies
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml]: @tomaszgy Change to `install mysql` (or `install mysql client`) to be more specific?

File: playbooks/roles/learner-profile-dashboard/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    name: "{{ LPD_LOG_DIR }}"
+    state: directory
+    owner: "{{ LPD_USER_NAME }}"
+
+- include: deploy-app.yml
+- include: deploy-gunicorn.yml
+- include: configure_nginx.yml
+# - include: log_volume.yml
+# - include: configure_log_download.yml
+#
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/tasks/main.yml]: @tomaszgy Remove or uncomment :)

File: playbooks/roles/learner-profile-dashboard/templates/lpd.site [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    ssl on;
+    ssl_certificate /etc/ssl/certs/ssl-cert.pem;
+    ssl_certificate_key /etc/ssl/private/ssl-cert.key;
+
+    server_name _;
+    return 404;
+}
+
+
+
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/templates/lpd.site]: @tomaszgy Let's make vertical whitespace consistent in this file.

File: playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+DB_PASSWORD = u'{{ LPD_DB_PASSWORD }}'
+DB_HOST = u'{{ LPD_DB_HOST }}'
+
+DEBUG = False
+
+ALLOWED_HOSTS = ['{{ LPD_SERVER_DOMAIN }}']
+
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_SCHEME', 'https')
+
+
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py]: @tomaszgy Extra blank line that we can remove.

File: playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+       'ENGINE': 'django.db.backends.mysql',
+       'NAME': DB_NAME,
+       'USER': DB_USERNAME,
+       'PASSWORD': DB_PASSWORD,
+       'HOST': DB_HOST
+   }
+}
+
+if getpass.getuser() == '{{ LPD_USER_NAME }}':
+    # change log paths only for production requests
```
Comment (itsjeyd) [inline playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py]: @tomaszgy Nit: Capitalize "change"

File: playbooks/roles/learner-profile-dashboard/defaults/main.yml (hunk 2 above)
Comment (tomaszgy) [reply playbooks/roles/learner-profile-dashboard/defaults/main.yml]: 👍 Removed.

File: playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py (hunk 9 above)
Comment (tomaszgy) [reply playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py]: 👍

File: playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py (hunk 8 above)
Comment (tomaszgy) [reply playbooks/roles/learner-profile-dashboard/templates/lpd_settings.py]: 👍

File: playbooks/roles/learner-profile-dashboard/defaults/main.yml (hunk 1 above)
Comment (tomaszgy) [reply playbooks/roles/learner-profile-dashboard/defaults/main.yml]: @itsjeyd yeah, I reused dalite ansible role to create that. This variable [is used by Gunicorn](http://docs.gunicorn.org/en/stable/run.html#django) and I guess that what the author of the comment meant was that this variable shouldn't be removed out of `LPD_ENV` (as `LPD_ENV` is later on passed to Gunicorn). Shall we reword it to `leave DJANGO_SETTINGS_MODULE inside LPD_NV`?

File: playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml (hunk 3 above)
Comment (tomaszgy) [reply playbooks/roles/learner-profile-dashboard/tasks/deploy-app.yml]: @itsjeyd, please see the current version of the comment.

File: playbooks/roles/learner-profile-dashboard/defaults/main.yml (hunk 1 above)
Comment (tomaszgy) [reply playbooks/roles/learner-profile-dashboard/defaults/main.yml]: @itsjeyd I went ahead and reworded it to that but will revert it if it doesn't make sense - please let me know :)

**Comment by tomaszgy** [comment]
@itsjeyd Just addressed your comments, please have a look :)

File: playbooks/roles/learner-profile-dashboard/defaults/main.yml (hunk 1 above)
Comment (itsjeyd) [reply playbooks/roles/learner-profile-dashboard/defaults/main.yml]: @tomaszgy That makes sense, thanks :)

**Comment by tomaszgy** [comment]
@itsjeyd thanks for review, merging now :)
